core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -274,8 +274,36 @@ private[spark] class DAGScheduler(
       accumUpdates: Seq[AccumulatorV2[_, _]],
       metricPeaks: Array[Long],
       taskInfo: TaskInfo): Unit = {
+    // It may cause heavy full GC problem if some heavy external accumulators keep in memory.
+    // We update these heavy accumulators before they entering into Spark listener event loop.
+    val (heavyAccumUpdates, otherAccumUpdates) = accumUpdates.partition { acc =>
```

Comment:
"Heavy" means that accumulated values are too big? If so, we cannot just check the size here and pre-merge it if the size goes over a threshold? IMHO it is not a good idea for developers to control the accumulator behavior explicitly via the name prefix.

Reply:
Yes. "Heavy" means the size is too big, such as the accumulator value contains thousands of file paths. I know this looks not a good solution. Do you mean to use a size threshold to filter out the big accumulator values is a better solution? The problem that way is we split the same accumulators (same name with different sizes). Actually, Spark uses METRICS_PREFIX to distinguish the internal metrics from all accumulators `val METRICS_PREFIX = "internal.metrics."`. We can split to `val (externalAccumUpdates, internalAccumUpdates)` and update/merge the `externalAccumUpdates` before they entering into event loop. But, you know, we still cannot forbid developers to use the "internal.metrics." prefix. We only could add some comments in code and document it.

Reply:
internal configs and user facing are completely different.  Yes if someone reads the code they could create internal ones but that is use at your own risk and we don't guarantee any api. This is user facing api essentially.  I agree with @maropu I'm not really fond of the prefix approach.   Is this only for CollectionAccumulator or SetAccumulator? Although it looks like the SetAccumulator is specific to delta.